### PR TITLE
Group framework dependent configuration

### PR DESCRIPTION
**Pull request by Corniel**
To increase the readability of the csproj file, I've grouped toe framework dependent configuration. I also added some `Label`'s.

### CODE REVIEW

**Review by jnyrup — COMMENTED** [review]
For the `<Compile Remove`, e.g. for `SystemExtensions.cs` I prefer the current feature-grouped approach as that'll ensure that future included TFMs don't include it.

But I have no strong opinions about this. I find both TFM and feature grouped approaches equally readable, it boils down to if you're current view is a feature or a TFM.

**Comment by dennisdoomen** [comment]
> But I have no strong opinions about this. I find both TFM and feature grouped approaches equally readable, it boils down to if you're current view is a feature or a TFM

I had the same feeling about this.

**Review by dennisdoomen — APPROVED** [review]

**Comment by Corniel** [comment]
I tried to get my head around what was disabled and when, And only be rewriting it, it started making sense.

**Comment by dennisdoomen** [comment]
On second thought, I actually think the result of this PR makes it better.

**Comment by Corniel** [comment]
@dennisdoomen 

If I understand correctly, the reason the Qodana steps fails is this one?

> To continue using Qodana, please ensure you have an access token and provide the token as the QODANA_TOKEN environment variable.

File: Src/FluentAssertions/FluentAssertions.csproj [hunk 1]
```diff
@@ -40,58 +43,82 @@
       <_Parameter1>Benchmarks, PublicKey=00240000048000009400000006020000002400005253413100040000010001002d25ff515c85b13ba08f61d466cff5d80a7f28ba197bbf8796085213e7a3406f970d2a4874932fed35db546e89af2da88c194bf1b7f7ac70de7988c78406f7629c547283061282a825616eb7eb48a9514a7570942936020a9bb37dca9ff60b778309900851575614491c6d25018fadb75828f4c7a17bf2d7dc86e7b6eafc5d8f</_Parameter1>
     </AssemblyAttribute>
   </ItemGroup>
-  <ItemGroup>
-    <None Remove="BannedSymbols.txt" />
```
Comment (jnyrup) [inline Src/FluentAssertions/FluentAssertions.csproj:44]: I can't recall why I added this.
Does removing it changes anything?
I.e. is it excluded from the nuget package and does the analyzer still flag usage of banned APIs?

**Comment by dennisdoomen** [comment]
> If I understand correctly, the reason the Qodana steps fails is this one?

We do have a valid license, but Github won't pass that token to PR builds. That's something we still need to choose a workaround for.

**Review by jnyrup — APPROVED** [review]

**Pull request merged**